CMakeLists.txt

```diff
@@ -262,6 +262,12 @@ endif()
 # =============================================================================
 find_package(PythonInterp REQUIRED)
 if(NRN_ENABLE_PYTHON)
+  # start afresh with PythonLibsNew's find_library
+  # without this, -DPYTHON_EXECUTABLE=`which python3` -DNRN_PYTHON_DYNAMIC="python3;python2"
+  # would end up with a NRN_DEFAULT_PYTHON_LIBRARIES of
+  # /usr/lib/x86_64-linux-gnu/libpython2.7.so
+  unset(PYTHON_LIBRARY CACHE)
```

Comment:
this can be set in `PythonLibsNew` module itself ? https://github.com/neuronsimulator/nrn/blob/5feec9097814717d159225ddcdfb7bb8ab31f36a/cmake/modules/FindPythonLibsNew.cmake#L52

Reply:
That makes a lot of sense.  I just didn't know if that was to be treated as sacrosanct. There is a similar unset in PythonDynamicHelper.cmake (in fact two, but I still don't understand the need in that context for ```unset(PYTHON_INCLUDE_DIR CACHE)``` and I left it out with respect to my change.)  By the way,  reading the cmake
documentation about ```find_library```, I didn't see any mention of it leaving ```NAME``` unchanged if it was already set.

Reply:
> That makes a lot of sense. I just didn't know if that was to be treated as sacrosanct.

It's copied from pybind11 project and ideally kept unchanged. But can be tweaked if we see further changes are required.

In case of NEURON we do multiple things : find interpreter, find library or sometime find multiple pythons. In these all scenarios, I think there is some inconsistent behaviour.

> There is a similar unset in PythonDynamicHelper.cmake (in fact two, but I still don't understand the need in that context for unset(PYTHON_INCLUDE_DIR CACHE) and I left it out with respect to my change.)

As we find multiple python versions in PythonDynamicHelper.cmake, `unset` were used bit experimentally. Ideally one should unset everything but I found minimal set to `unset` so that next python version was found without any side effects.

Reply:
> ideally kept unchanged. But can be tweaked

I guess my judgement is slightly in favor of keeping the pybind11 copy unchanged.  It seems easier to occasionally compare copies of what is in nrn/cmake/modules with their origin if ours remain unchanged as much as possible.